lib/tasks/one_off.rake

```diff
@@ -259,8 +259,9 @@ namespace :one_off do
   end
 
   desc "Delete ignored Maven versions and resync packages"
-  task :delete_ignored_maven_versions_and_resync_packages, %i[commit] => :environment do |_t, args|
-    commit = args.commit.present? && args.commit == "yes"
+  task :delete_ignored_maven_versions_and_resync_packages, %i[max commit] => :environment do |_t, args|
+    max = args[:max].presence.to_i == 0 ? nil : args[:max].to_i
```

Comment:
There are two optional arguments. To handle the case where we want the `max` missing but the `commit` present, I'm just treating the first one here so that a non-numerical value will mean it defaults to not being used.